ttljob: implement a delete based rate limit

otan opened the pull request:
(only last commit matters)

This commit introduces a delete based rate limit for TTL. It can be
configured by the `ttl_delete_rate_limit` storage parameter, but
defaults to the value in `sql.ttl.default_delete_rate_limit` (if zero,
assume no rate limit).

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/76615)
<!-- Reviewable:end -->

rafiss commented on pkg/sql/ttl/ttljob/ttljob.go:
nit: maybe ttl-delete

otan committed:
ttljob: implement a delete based rate limit

This commit introduces a delete based rate limit for TTL. It can be
configured by the `ttl_delete_rate_limit` storage parameter, but
defaults to the value in `sql.ttl.default_delete_rate_limit` (if zero,
assume no rate limit).

Release note: None

otan commented:
bors r=rafiss

otan commented:
bors r=rafiss
